src/reference/antora/modules/ROOT/pages/mail.adoc

```diff
@@ -515,15 +1040,114 @@ In other words, while our adapter may peek at the email, it also lets the email
 You can send the messages to some other Spring Integration components or invoke a custom bean to perform some action.
 For example, to move an IMAP message to a different folder after the transaction commits, you might use something similar to the following:
 
-[source,xml]
+[tabs]
+======
+Java::
++
+[source,java,role="primary"]
+----
+@Bean
+TransactionSynchronizationFactory transactionSynchronizationFactory() {
+    SpelExpressionParser parser = new SpelExpressionParser();
+    ExpressionEvaluatingTransactionSynchronizationProcessor expressionEvaluatingTransactionSynchronizationProcessor =
+            new ExpressionEvaluatingTransactionSynchronizationProcessor();
+    expressionEvaluatingTransactionSynchronizationProcessor.setAfterCommitExpression(parser.parseExpression("@syncProcessor.process(payload)"));
+    return new DefaultTransactionSynchronizationFactory(expressionEvaluatingTransactionSynchronizationProcessor);
+}
+
+@Bean
+public ImapIdleChannelAdapter imapIdleChannelAdapter(ImapMailReceiver imapMailReceiver, MessageChannel receiveChannel,
+TransactionSynchronizationFactory transactionSynchronizationFactory) {
+
+    ImapIdleChannelAdapter adapter = new ImapIdleChannelAdapter(imapMailReceiver);
+    adapter.setOutputChannel(receiveChannel);
+    adapter.setAutoStartup(true);
+    adapter.setTransactionSynchronizationFactory(transactionSynchronizationFactory);
+    return adapter;
+}
+----
+Java DSL::
++
+[source,java,role="secondary"]
+----
+@Bean
+public IntegrationFlow imapIdleFlow(ImapMailReceiver imapMailReceiver, MessageChannel receiveChannel,
+    MailMessageHandler mailMessageHandler) {
+
+	SpelExpressionParser parser = new SpelExpressionParser();
+	ExpressionEvaluatingTransactionSynchronizationProcessor expressionEvaluatingTransactionSynchronizationProcessor =
+        new ExpressionEvaluatingTransactionSynchronizationProcessor();
+	expressionEvaluatingTransactionSynchronizationProcessor.setAfterCommitExpression(parser.parseExpression("@syncProcessor.process(payload)"));
+	var transactionSynchronizationFactory = new DefaultTransactionSynchronizationFactory(expressionEvaluatingTransactionSynchronizationProcessor);
+
+    return IntegrationFlow
+        .from(Mail.imapIdleAdapter(imapMailReceiver)
+            .autoStartup(true)
+            .id("imapIdleAdapter")
+                .transactionSynchronizationFactory(transactionSynchronizationFactory))
+        .channel(receiveChannel)
+        .handle(mailMessageHandler)
+        .get();
+}
+----
+Kotlin DSL::
++
+[source,kotlin,role="secondary"]
+----
+@Bean
+fun imapIdleFlow(imapMailReceiver: ImapMailReceiver, receiveChannel: MessageChannel,
+    mailMessageHandler: MailMessageHandler): IntegrationFlow {
+    val processor = ExpressionEvaluatingTransactionSynchronizationProcessor().apply {
+        setAfterCommitExpression(
+            SpelExpressionParser().parseExpression("@syncProcessor.process(payload)")
+        )
+    }
+
+    return IntegrationFlow
```

Comment:
Kotlin DSL sample needs more work.
And `ExpressionEvaluatingTransactionSynchronizationProcessor` has to be a bean.